packages/tools/client-debugger/client-debugger-view/src/components/TelemetryView.tsx

```diff
@@ -29,16 +32,16 @@ export function TelemetryView(): React.ReactElement {
 		};
 
 		// Event handler for messages coming from the Message Relay
-		function messageHandler(event: MessageEvent<Partial<IDebuggerMessage>>): void {
-			handleIncomingMessage(event.data, inboundMessageHandlers);
+		function messageHandler(message: Partial<IDebuggerMessage>): void {
+			handleIncomingMessage(message, inboundMessageHandlers);
 		}
 
-		globalThis.addEventListener("message", messageHandler);
+		messageRelay.on("message", messageHandler);
 
 		return (): void => {
-			globalThis.removeEventListener("message", messageHandler);
+			messageRelay.off("message", messageHandler);
 		};
-	}, [telemetryEvents, setTelemetryEvents]);
+	}, [messageRelay, telemetryEvents, setTelemetryEvents]);
```

Comment:
So we ran into a scenario where we know better than the linter and will have to disable the rule? Living on the edge 😝 . But I do get your point.

Reply:
Oh, I missed that we were actually observing the state of `telemetryEvents` by concatenating the existing ones with the new ones coming in. I think we actually **do** need to list it here, otherwise we could end up concatenating with the wrong `telemetryEvents` state (I think).

This might be a case where we need to wrap the concat logic in a `useCallback` hook, to ensure we're always getting the most up-to-date version of `telemetryEvents`. I'll admit, I don't have a lot of experience with this particular use case, so this might require a bit of investigation.

Reply:
[This](https://react.dev/reference/react/useEffect#updating-state-based-on-previous-state-from-an-effect) was the solution. `useCallback` would have had the same problem, I think. The callback function itself would need to be passed as a dependency to the effect, and we end up with the same cycle but in two steps instead of one.

Reply:
Ah, yeah that would do it!